docs/master-server.md

```diff
@@ -24,6 +26,34 @@ to complete the update. All Salt states shown should be green.
 
 ## updating the master server
 
-Whenever a change is made to the formulas or to your builder-private repository, the master server needs to be updated. It will do so every hour through a cron job, but you can always do this immediately after you want a change to be applied with:
+Whenever a change is made to the formulas, or to your `builder-private` repository, or a minion is destroyed, the master server needs to be updated. It will do so every hour through a cron job but can also be run immediately with:
+
+	./bldr master.update:master-server--prod
+	
+## replacing the master server
+
+The first step in replacing a master server is to launch a new instance. 
+
+In the presence of multiple master servers, `builder` will sort them alphanumerically and prefer the first instance it finds. `master-server--a` would take precendence over `master-server--b` and `master-server--2001-01-01` would take precendence over `master-server--2012-12-12`.
+
+All existing minions have the address for their master-server encoded in their `context` data and `build_vars`. The process of getting an existing minion to talk to a new master is called *re-mastering*.
+
+Once the new master server has been properly created and configured it should be ready to command new minions out of the box.
+
+To re-master a single minion:
+
+    ./bldr master.remaster:project--ci,master-server--newinstanceid
+    
+To re-master *all* minions:
+
+    ./bldr master.remaster_all:master-server--newinstanceid
+    
+The process of re-mastering a minion roughly follows:
+
+* replace the value of `aws.ec2.master_ip` in the `context` data
```

Comment:
still has the problem that if someone runs `update_infrastructure`, the `context` is regenerated and the switch is done. So a preliminary step here should be to do what we did: pin the new master ip address in `projects/elife.yaml`. Then remastering is a quicker switch _towards what is already pinned there_, hence it won't be rolled back.